templates/slurm/footer_tnt.ftl

```diff
@@ -3,26 +3,28 @@ mydate_finished=$(date +"%Y-%m-%dT%H:%M:%S+0200")
 <#noparse>
 if curl -f -s -H "Content-Type: application/json" -X POST -d "{"username"="${USERNAME}", "password"="${PASSWORD}"}" https://${MOLGENISSERVER}/api/v1/login
 then
-	CURLRESPONSE=$(curl -H "Content-Type: application/json" -X POST -d "{"username"="${USERNAME}", "password"="${PASSWORD}"}" https://${MOLGENISSERVER}/api/v1/login)
-	TOKEN=${CURLRESPONSE:10:32}
-
-	if curl -f -s -H "Content-Type:application/json" -H "x-molgenis-token:${TOKEN}" -X PUT -d "finished" https://${MOLGENISSERVER}/api/v1/status_jobs/</#noparse>${project}_${taskId}/status
-	then
-		echo "set"
-	else
-		echo "not set"
-	fi
-
-	<#noparse>
-	sleep 1
-	if curl -f -s -H "Content-Type:application/json" -H "x-molgenis-token:${TOKEN}" -X PUT -d "'${mydate_finished}'" https://${MOLGENISSERVER}/api/v1/status_jobs/</#noparse>${project}_${taskId}/finished_date
+	if CURLRESPONSE=$(curl -H "Content-Type: application/json" -X POST -d "{"username"="${USERNAME}", "password"="${PASSWORD}"}" https://${MOLGENISSERVER}/api/v1/login)
 	then
-	        echo "set"
+		TOKEN=$(echo "${CURLRESPONSE}" | awk 'BEGIN {FS=":"} $1 ~ /token/ {print $2}' | awk 'BEGIN {FS="\""}{print $2}')
+		if curl -f -s -H "Content-Type:application/json" -H "x-molgenis-token:${TOKEN}" -X PUT -d "finished" https://${MOLGENISSERVER}/api/v1/status_jobs/</#noparse>${project}_${taskId}/status
+		then
+			echo "set"
+		else
+			echo "not set"
+		fi
+
+		<#noparse>
+		sleep 1
+		if curl -f -s -H "Content-Type:application/json" -H "x-molgenis-token:${TOKEN}" -X PUT -d "'${mydate_finished}'" https://${MOLGENISSERVER}/api/v1/status_jobs/</#noparse>${project}_${taskId}/finished_date
+		then
+			echo "set"
+		else
+			echo "not set"
+		fi
 	else
-        	echo "not set"
+		echo "FATAL: CURLRESPONSE"
```

Comment:
FATAL -> ERROR. Als het FATAL zou zijn, dan had er een exit 1 o.i.d achter gemoeten en stopt je script. Dat is nu net niet de bedoeling.

Reply:
ik was dit nog aan het testen :)

Reply:
Misschien geeft cURL nog een relevante ERROR terug; dan is het wel zo handig om die te loggen. Nu wordt alleen de letterlijke tekst CURLRESPONSE gelogd; daar hebben we niet zo veel aan. Het kan ook zijn dat de exit state van cURL niet nul is en er geen waarde terug komt en de variabele leeg blijft. Om te voorkomen dat de boel dan alsnog crasht door set -u zou het zoiets moeten zijn:
`echo "ERROR: cURL failed with: ${CURLRESPONSE-:}"`
Tevens moet je dan wel STDERR naar STDOUT redirecten met `2>&1` bij het doen van cURL. Dat ontbreekt nu nog. Tot slot zitten in die cURL unescaped quotes op een verkeerde plek en ontbreken op andere plaatsen quotes. Het zou zoiets moeten worden:
`if CURLRESPONSE="$(curl -f -s -H "Content-Type: application/json" -X POST -d "{\"username\"=\"${USERNAME}\", \"password\"=\"${PASSWORD}\"}" "https://${MOLGENISSERVER}/api/v1/login" 2>&1)"`

Reply:
> ik was dit nog aan het testen :)

Ok, voor de volledigheid: Dezelfde problemen als in de footer zitten ook nog in de header.